src/ess/polarization/supermirror.py

```diff
@@ -51,6 +52,47 @@ def __call__(self, *, wavelength: sc.Variable) -> sc.DataArray:
         )
 
 
+@dataclass
+class EfficiencyLookupTable(SupermirrorEfficiencyFunction[PolarizingElement]):
+    """
+    Efficiency of a supermirror as a lookup table.
+    The names of the columns in the table has to be "wavelength", "efficiency".
+
+    Parameters
+    ----------
+    table:
+        The lookup table.
+    """
+
+    table: sc.DataArray
+
+    def __call__(self, *, wavelength: sc.Variable) -> sc.DataArray:
+        """Return the efficiency of a supermirror for a given wavelength"""
+        return sc.lookup(sc.values(self.table), 'wavelength')(wavelength)
```

Comment:
- Not sure how costly this is, but the lookup could be created on init, to keep this cheap in stream processing.
- Is `sc.lookup` already constructing a `DataArray`, or do we need to do so ourselves?